homeassistant/components/camera/dlink.py

```diff
@@ -0,0 +1,341 @@
+        if (self._web_ui_form_data['motion']['MotionDetectionBlockSet'] ==
+                '0000000000000000000000000'):
+            self._web_ui_form_data['motion']['MotionDetectionBlockSet'] = (
+                '1111111111111111111111111')
+        requests.post(
+            self.base_url + 'setSystemMotion',
+            data=self._web_ui_form_data['motion'],
+            auth=(self.username, self.password))
+
+        # self._is_motion_detection_enabled = False
+        self.refesh_all_settings_from_device()
+        self.update_ha_state(True)
+
+        # self.send_motion_state()
+
+        return True
+
+    def disable_motion_detection(self):
+        """ Disable the motion detection settings for the camera. """
+        # pylint: disable=missing-super-argument
+        can_enable = super().disable_motion_detection()
+        if can_enable is False:
+            return can_enable
+
+        self._web_ui_form_data['motion']['MotionDetectionEnable'] = 0
+        requests.post(
+            self.base_url + 'setSystemMotion',
+            data=self._web_ui_form_data['motion'],
+            auth=(self.username, self.password))
+
+        self.refesh_all_settings_from_device()
+        self.update_ha_state(True)
+
+        # self.send_motion_state()
+
+        return True
+
+    def set_ftp_details(self):
+        """ Sets the FTP details used by the camera to upload motion
+            detected images.  The details will be set to the appropriate
+            values from the loaded ftp component """
+        # pylint: disable=missing-super-argument
+        can_enable = super().set_ftp_details()
+        if can_enable is False:
+            return can_enable
+
+        ftp_server = get_component('ftp').FTP_SERVER
+
+        ftp_path = self.ftp_path
+
+        if not os.path.isdir(ftp_path):
+            os.makedirs(ftp_path)
+            _LOGGER.info(
+                'Camera %s image path did not exist and was \
+                atomatically created at %s',
+                self.entity_id,
+                ftp_path)
+
+        self._web_ui_form_data['upload']['FTPHostAddress'] = (
+            ftp_server.server_ip)
+        self._web_ui_form_data['upload']['FTPPortNumber'] = (
+            ftp_server.server_port)
+        self._web_ui_form_data['upload']['FTPUserName'] = (
+            ftp_server.username)
+        self._web_ui_form_data['upload']['FTPPassword'] = (
+            ftp_server.password)
+        self._web_ui_form_data['upload']['FTPDirectoryPath'] = (
+            self.entity_id)
+
+        self._web_ui_form_data['upload']['FTPScheduleEnable'] = '1'
+        self._web_ui_form_data['upload']['FTPScheduleMode'] = '2'
+
+        requests.post(
+            self.base_url + 'setSystemFTP',
+            data=self._web_ui_form_data['upload'],
+            auth=(self.username, self.password))
+
+        self.refesh_all_settings_from_device()
+        self.update_ha_state(True)
```

Comment:
I would imagine that the code calling `set_ftp_details` would be in charge of updating the state. That way there is no need to duplicate this call in each implementation (and maybe even forget it somewhere).